### ISSUE TITLE
`numba<0.56.0` in setup.py causes NumPy version conflict

### ISSUE THREAD

**Issue by 3tothe6** [SeldonIO/alibi-detect#647]
I ran `pip install alibi-detect[torch] && python -c "from alibi_detect.datasets import fetch_ecg"` in a newly created venv virtual environment on a Windows machine and got:
```
[PRIVATE INFORMATION]\venv\lib\site-packages\alibi_detect\utils\pytorch\kernels.py:60: UserWarning: Failed to initialize NumPy: module compiled against API version 0x10 but this version of numpy is 0xf (Triggered internally at  ..\torch\csrc\utils\tensor_numpy.cpp:68.)
```
It seems that PyTorch needs numpy==1.23.* but alibi-detect uses numba 0.55.2 and numba 0.55.2 requires numpy<1.23, therefore causing conflict. Merging #573 may solve this problem.

**Comment by 3tothe6** [comment]
* My Python version is 3.10.7.
* Outputs of `pip install alibi-detect[torch]` and `pip list` can be seen [here](https://github.com/SeldonIO/alibi-detect/files/9750537/pip-install.log) and [here](https://github.com/SeldonIO/alibi-detect/files/9750548/pip-list.log) respectively. `pip check` says `No broken requirements found.`. I ran these pip commands within a new environment simply created by `python -m venv venv`.
* Running `python -c "from alibi_detect.datasets import fetch_ecg"` after the above commands got:

```
E:\venv\lib\site-packages\alibi_detect\utils\pytorch\kernels.py:60: UserWarning: Failed to initialize NumPy: module compiled against API version 0x10 but this version of numpy is 0xf (Triggered internally at  ..\torch\csrc\utils\tensor_numpy.cpp:68.)
  self.log_sigma = nn.Parameter(torch.empty(1), requires_grad=trainable)
```
> so would like to look into the order in which numpy was installed and uninstalled in your env

NumPy was installed only once, as in the above log.

**Comment by 3tothe6** [comment]
I cloned this repo and switched to v0.10.3, removed the `numba<0.56.0` constraint and did the same things above but install alibi-detect from my modified version, then the conflict disappeared.

**Comment by ascillitoe** [comment]
Hi @3tothe6, thank you for the above. Looking at your pip logs it looks like `numpy 1.22` was collected but then replaced by `numpy 1.23` after `numba` was collected. This and your later experiment seem to support your initial assertion. We will likely merge #573 today, and it will be included in the incoming v0.11.0 release. Thanks again!
